<a href="https://colab.research.google.com/github/Auzek2002/Data_Science_101/blob/main/Assignment_03.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

1) A company decided to give bonus of 5% to employee if his/her year of service is more than 5 years.
Ask user for their salary and year of service and print the net bonus amount.

In [3]:
#Type your code here
sal = int(input("Enter salary: "))
year = int(input("Enter years of service: "))
if year > 5:
  b = sal*(5/100)
  sal += b
  print(f"Salary with bonus: {sal}")
else:
  print(f"Salary: {sal}")

Enter salary: 20000
Enter years of service: 3
Salary: 20000



2) Write a program to check whether a person is eligible for voting or not. (accept age from user) if age is greater than 17 eligible otherwise not eligible

In [5]:
#Type your code here
age = int(input("Age: "))
if age > 17:
  print("Eligible")
else:
  print("Not Eligible")

Age: 18
Eligible



3) Write a program to check whether a number entered by user is even or odd.

In [8]:
#Type your code here
num = int(input("Enter a number: "))
if num%2==0:
  print(f"{num} is Even!")
else:
  print(f"{num} is Odd!")

Enter a number: 4
4 is Even!


4) Write a program to check whether a number is divisible by 7 or not.
Show Answer

In [9]:
#Type your code here
num = int(input("Enter a number: "))
if num%7==0:
  print(f"{num} is Divisible!")
else:
  print(f"{num} is not Divisible!")

Enter a number: 14
14 is Divisible!


5) Write a program to display
"Hello" if a number entered by user is a multiple of five , otherwise print "Bye".

In [10]:
#Type your code here
num = int(input("Enter a number: "))
if num%5==0:
  print("Hello!")


Enter a number: 10
Hello!


7) Write a program to display the last digit of a number.


In [12]:
#Type your code here
num = int(input("Enter a number: "))
if num<10:
  print(num)
else:
  print(num%10)

Enter a number: 24
4


9) Take values of length and breadth of a rectangle from user and print if it is square or rectangle.

In [13]:
#Type your code here
l = float(input("Enter the length: "))
b = float(input("Enter the breadth: "))
if l == b:
  print("Sqaure")
else:
  print("Rectangle")

Enter the length: 12
Enter the breadth: 2
Rectangle


10) Take two int values from user and print greatest among them.

In [14]:
# Type your code here
a = int(input("Enter the 1st value: "))
b = int(input("Enter the 2nd value: "))
if a>b:
  print(f"{a} is the greatest!")
else:
  print(f"{b} is the greatest!")

Enter the 1st value: 2
Enter the 2nd value: 3
3 is the greatest!


11) A shop will give discount of 10% if the cost of purchased quantity is more than 1000.
Ask user for quantity
Suppose, one unit will cost 100.
Judge and print total cost for user.

In [15]:
#Type your code here
q = int(input("Enter the amount of things purchased: "))
p = q* 100
if p>1000:
  p*=(10/100)
  print(f"Price after discount is: {p}")
else:
  print(f"Price is: {p}")


Enter the amount of things purchased: 150
Price after discount is: 1500.0


12) A school has following rules for grading system:

a. Below 25 - F

b. 25 to 45 - E

c. 45 to 50 - D

d. 50 to 60 - C

e. 60 to 80 - B

f. Above 80 - A

Ask user to enter marks and print the corresponding grade.

In [16]:
#Type your code here
marks = int(input("Enter your marks: "))
if marks < 25:
  print('F')
if marks > 25 and marks < 45:
  print('E')
if marks > 45 and marks < 50:
  print('D')
if marks > 50 and marks < 60:
  print('C')
if marks > 60 and marks < 80:
  print('B')
if marks > 80:
  print('A')

Enter your marks: 100
A


14)A student will not be allowed to sit in exam if his/her attendence is less than 75%.

Take following input from user

Number of classes held

Number of classes attended.

And print

percentage of class attended

Is student is allowed to sit in exam or not.


In [20]:
#Type your code here
c = int(input("Enter number of classes held: "))
a = int(input("Enter the number of classes attended: "))
p = (a/c)*100
print(f"Percentage of class attended: {p:0.1f}")
if p > 75:
  print("Student is allowed to sit in the exam")
else:
  print("Student is not allowed to sit in the exam")

Enter number of classes held: 100
Enter the number of classes attended: 76
Percentage of class attended: 76.0
Student is allowed to sit in the exam


15) Modify the above question to allow student to sit if he/she has medical cause. Ask user if he/she has medical cause or not ( 'Y' or 'N' ) and print accordingly.

In [22]:
#Type your code here
c = int(input("Enter number of classes held: "))
a = int(input("Enter the number of classes attended: "))
p = (a/c)*100
print(f"Percentage of class attended: {p:0.1f}")
if p > 75:
  print("Student is allowed to sit in the exam")
else:
  m = input("Did you have a medical cause? Y if yes N if no: ")
  if m == 'Y':
    print("Student is allowed to sit in the exam")
  else:
    print("Student is not allowed to sit in the exam")

Enter number of classes held: 100
Enter the number of classes attended: 71
Percentage of class attended: 71.0
Did you have a medical cause? Y if yes N if no: N
Student is not allowed to sit in the exam


16) Write a program to check if a year is leap year or not.

If a year is divisible by 4 then it is leap year but if the year is century year like 2000, 1900, 2100 then it must be divisible by 400.

In [23]:
#Type your code here
year = int(input("Enter a year: "))

if (year % 4 == 0 and year % 100 != 0) or (year % 400 == 0):
    print(f"{year} is a leap year.")
else:
    print(f"{year} is not a leap year.")


Enter a year: 2014
2014 is not a leap year.


17) Ask user to enter age, gender ( M or F ), marital status ( Y or N ) and then using following rules print their place of service.

if employee is female, then she will work only in urban areas.

if employee is a male and age is in between 20 to 40 then he may work in anywhere

if employee is male and age is in between 40 t0 60 then he will work in urban areas only.

And any other input of age should print "ERROR"

In [24]:
#Type your code here
age = int(input("Enter age: "))
gender = input("Enter gender (M or F): ")
marital_status = input("Enter marital status (Y or N): ")

if age < 20 or age > 60:
    print("ERROR")
elif gender == 'F':
    print("Urban areas")
elif gender == 'M':
    if 20 <= age <= 40:
        print("Anywhere")
    elif 40 <= age <= 60:
        print("Urban areas only")


Enter age: 33
Enter gender (M or F): M
Enter marital status (Y or N): Y
Anywhere


6) Write a program to calculate the electricity bill (accept number of unit from user) according to the following criteria :
     Unit                                                     Price  
uptp 100 units                                             no charge
Next 200 units                                              Rs 5 per unit
After 200 units                                             Rs 10 per unit
(For example if input unit is 350 than total bill amount is Rs.3500
(For example if input unit is 97 than total bill amount is Rs.0
(For example if input unit is 150 than total bill amount is Rs.750

In [28]:
#Type your code here
units = int(input("Enter the number of units: "))
total_bill = 0

if units <= 100:
    total_bill = 0
elif units <= 300:
    total_bill = (units - 100) * 5
elif units > 300:
    total_bill = 200 * 5 + (units - 300) * 10

print(f"Total bill amount is Rs. {total_bill}")


Enter the number of units: 97
Total bill amount is Rs. 0


13) Take input of age of 3 people by user and determine oldest and youngest among them.

In [29]:
#Type your code here
a1 = int(input("Enter the age of person 1: "))
a2 = int(input("Enter the age of person 2: "))
a3 = int(input("Enter the age of person 3: "))

if a1> a2 and a1>a3:
  print("Person 1 is the oldest")
if a2> a1 and a2>a3:
  print("Person 2 is the oldest")
if a3> a2 and a3>a1:
  print("Person 3 is the oldest")

if a1< a2 and a1<a3:
  print("Person 1 is the youngest")
if a2< a1 and a2<a3:
  print("Person 2 is the youngest")
if a3< a2 and a3<a1:
  print("Person 3 is the youngest")

Enter the age of person 1: 12
Enter the age of person 2: 30
Enter the age of person 3: 20
Person 2 is the oldest
Person 1 is the youngest
